source/Calamari/Kubernetes/Commands/KubernetesApplyRawYamlCommand.cs

```diff
@@ -73,52 +62,58 @@ public override int Execute(string[] commandLineArguments)
-            var configurationTransformer = ConfigurationTransformer.FromVariables(variables, log);
-            var transformFileLocator = new TransformFileLocator(fileSystem, log);
-            var replacer = new ConfigurationVariablesReplacer(variables, log);
             var conventions = new List<IConvention>
             {
-                new DelegateInstallConvention(d =>
+                conventionFactory.ResolveDelegate(d =>
                 {
-                    extractPackage.ExtractToStagingDirectory(pathToPackage);
-                    //If using the inline yaml, copy it to the staging directory.
-                    var inlineFile = Path.Combine(Environment.CurrentDirectory, "customresource.yml");
-                    var stagingDirectory = Path.Combine(Environment.CurrentDirectory, "staging");
-                    if (fileSystem.FileExists(inlineFile))
+                    if (pathToPackage != null)
                     {
-                        fileSystem.CopyFile(inlineFile, Path.Combine(stagingDirectory, "customresource.yml"));
+                        extractPackage.ExtractToStagingDirectory(pathToPackage, workingDirectory: d.CurrentDirectory);
                     }
-                    d.StagingDirectory = stagingDirectory;
-                    d.CurrentDirectoryProvider = DeploymentWorkingDirectory.StagingDirectory;
+                    else
+                    {
+                        //If using the inline yaml, copy it to the staging directory.
+                        var inlineFile = Path.Combine(d.CurrentDirectory, "customresource.yml");
+                        var stagingDirectory = Path.Combine(d.CurrentDirectory, "staging");
+                        fileSystem.EnsureDirectoryExists(stagingDirectory);
+                        if (fileSystem.FileExists(inlineFile))
+                        {
+                            fileSystem.MoveFile(inlineFile, Path.Combine(stagingDirectory, "customresource.yml"));
+                        }
+                        d.StagingDirectory = stagingDirectory;
+                        d.CurrentDirectoryProvider = DeploymentWorkingDirectory.StagingDirectory;
+                    }
+
+                    kubectl.WorkingDirectory = d.CurrentDirectory;
+                    kubectl.EnvironmentVariables = d.EnvironmentVariables;
                 }),
-                new SubstituteInFilesConvention(new SubstituteInFilesBehaviour(substituteInFiles)),
-                new ConfigurationTransformsConvention(new ConfigurationTransformsBehaviour(fileSystem, variables, configurationTransformer, transformFileLocator, log)),
-                new ConfigurationVariablesConvention(new ConfigurationVariablesBehaviour(fileSystem, variables, replacer, log)),
-                new StructuredConfigurationVariablesConvention(new StructuredConfigurationVariablesBehaviour(structuredConfigVariablesService)),
+                conventionFactory.Resolve<SubstituteInFilesConvention>(),
```

Comment:
why is this command now using a convention factory rather than just new-ing it up like the others?

It might be an idea to consider how we can update how the relationship for _all_ commands and conventions work but its not clear why this one in particular needs to be different. Presumably for testing?

Reply:
I've reverted this after discussing with Mark.